packages/core/react/use-chat.ts

```diff
@@ -101,11 +101,20 @@ const getStreamedResponse = async (
   const constructedMessagesPayload = sendExtraMessageFields
     ? chatRequest.messages
     : chatRequest.messages.map(
-        ({ role, content, name, function_call, tool_calls, tool_call_id }) => ({
+        ({
+          role,
+          content,
+          data,
+          name,
+          function_call,
+          tool_calls,
+          tool_call_id,
+        }) => ({
           role,
           content,
           tool_call_id,
           ...(name !== undefined && { name }),
+          ...(data !== undefined && { data }),
```

Comment:
If we want to pass this on for messages, we'd need this to change in Svelte, Solid and Vue `useChat` as well.